configure.ac

```
@@ -11,6 +11,14 @@ AC_ARG_WITH([pytest],[  --with-pytest=executable],[PYTEST="$withval"])
 if test "x$PYTEST" = x; then
     AC_CHECK_PROGS([PYTEST],[pytest pytest-3],[pytest])
 fi
+
+case ${host_os} in
+    *solaris*)
+        build_solaris=yes ;;
+esac
+
+AM_CONDITIONAL([SOLARIS], [test "${build_solaris}" = yes])
```

```suggestion
AM_CONDITIONAL([SOLARIS], [test "x${build_solaris}" = xyes])
```

The environment might have the environment variable `build_solaris` starting with `-`.

---

This was barely relevant at all, since requiring a leading x only affects shells that aren't POSIX conformant, but AFAIK the last bug in a valid /bin/sh was fixed in 2008 and the last bug in an invalid /bin/sh (which is to say, zsh) was fixed in 2015.

---

The configure script (autoconf) traditionally tried to cover non-POSIX conformant shells as well. Prefixing `x` seems consistent with the autoconf convention. It might not be relevant in practical cases now, but is there a reason to intentionally drop non-POSIX shells, which autoconf has been supported? This is a small fix. Is there a reason to stick with the style which is inconsistent with the autoconf convention?

---

I mean I wouldn't try to argue that this would become a practical issue, but I should probably say it would be more like a style issue.

---

I fixed it by explicitly setting it equal to the blank string before the case statement. :D Environment variables are now entirely ignored.

---

Thanks. That works.

If it were me, I'd still follow the `x` style in the Autoconf configurations even when the variable value is ensured to be safe, but that should be a preference.

> which means you cannot confuse configure by running `build_solaris=yes ./configure`

Yeah, I assumed you intentionally left the possibility of that usage.

---

https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.72/html_node/Multiple-Cases.html

> Here is an example that uses the shell variable `fstype` to keep track of whether the remaining cases need to be checked. Note that since the value of `fstype` is under our control, we don’t have to use the longer ‘`test "x$fstype" = xno`’.

Seems like a suggestion that it's something beneficial to avoid... (for readability reasons it feels better to avoid IMHO)

---

> Yeah, I assumed you intentionally left the possibility of that usage.

In general one would do this via `./configure --host=XXX`.

---

> https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.72/html_node/Multiple-Cases.html
> 
> > Here is an example that uses the shell variable `fstype` to keep track of whether the remaining cases need to be checked. Note that since the value of `fstype` is under our control, we don’t have to use the longer ‘`test "x$fstype" = xno`’.
> 
> Seems like a suggestion that it's something beneficial to avoid...

Oh, I see. I didn't know that recommendation. Thank you.

> > Yeah, I assumed you intentionally left the possibility of that usage.
> 
> In general one would do this via `./configure --host=XXX`.

Yes.